lib/realms/zones/transfer.rb

```diff
@@ -17,6 +18,7 @@ def initialize(source:, destination:, card: source.first, destination_position:
 
       def transfer!
         raise InvalidTarget, card.key if !source.include?(card) || destination.include?(card)
+        @id = source.next_transfer_id
```

Comment:
does this really need to be assigned at `transfer!`? I think so because it's essentially equivalent to commit transaction. No reason other things couldn't do a transfer while a transfer is being created.